commit 250d1bd3f4ed0b50d79b3ec81ccefbabb203f916
Author: Jean Delvare <khali@linux-fr.org>
Date:   Sun Dec 10 21:21:33 2006 +0100

    i2c: Enable PEC on more i2c-i801 devices
    
    Enable PEC on recent Intel SMBus controllers (ICH6, ICH7, ICH8, ICH9
    and ESB2.)
    
    Signed-off-by: Jean Delvare <khali@linux-fr.org>
    Acked-by: Jason Gaston <jason.d.gaston@intel.com>

---
 drivers/i2c/busses/i2c-i801.c |   16 	12 +	4 -	0 !
 1 files changed, 12 insertions(+), 4 deletions(-)

Index: linux-2.6.17/drivers/i2c/busses/i2c-i801.c
===================================================================
--- linux-2.6.17.orig/drivers/i2c/busses/i2c-i801.c	2007-01-02 15:35:48.000000000 +0100
+++ linux-2.6.17/drivers/i2c/busses/i2c-i801.c	2007-01-02 15:36:16.000000000 +0100
@@ -122,12 +122,20 @@ static int i801_setup(struct pci_dev *de
 		return -ENODEV;
 
 	I801_dev = dev;
-	if ((dev->device == PCI_DEVICE_ID_INTEL_82801DB_3) ||
-	    (dev->device == PCI_DEVICE_ID_INTEL_82801EB_3) ||
-	    (dev->device == PCI_DEVICE_ID_INTEL_ESB_4))
+	switch (dev->device) {
+	case PCI_DEVICE_ID_INTEL_82801DB_3:
+	case PCI_DEVICE_ID_INTEL_82801EB_3:
+	case PCI_DEVICE_ID_INTEL_ESB_4:
+	case PCI_DEVICE_ID_INTEL_ICH6_16:
+	case PCI_DEVICE_ID_INTEL_ICH7_17:
+	case PCI_DEVICE_ID_INTEL_ESB2_17:
+	case PCI_DEVICE_ID_INTEL_ICH8_5:
+	case PCI_DEVICE_ID_INTEL_ICH9_6:
 		isich4 = 1;
-	else
+		break;
+	default:
 		isich4 = 0;
+	}
 
 	/* Determine the address of the SMBus areas */
 	if (force_addr) {
